File: fidouafclient/src/main/java/io/hanko/fidouafclient/utility/Preferences.java
package io.hanko.fidouafclient.utility;

import android.content.Context;
import android.content.SharedPreferences;

import java.util.HashSet;
import java.util.Set;

public class Preferences {

    public static String FINGERPRINT_PREFERENCE = "FidoASMPreferences";
    public static String LOCKSCREEN_PREFERENCE = "FidoLockscreenPreferences";

    public static SharedPreferences create(Context c, String name) {
        return c.getSharedPreferences(name, Context.MODE_PRIVATE);
    }

    public static void setParam(final SharedPreferences preferences, final String paramName, final String paramValue) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString(paramName, paramValue);
        editor.apply();
    }

    public static String getParam(final SharedPreferences preferences, final String paramName) {
        return preferences.getString(paramName, "");
    }

    public static Set<String> getParamSet(final SharedPreferences preferences, final String paramName) {
        return preferences.getStringSet(paramName, new HashSet<String>());
    }

    public static void setParamSet(final SharedPreferences preferences, final String paramName, final Set<String> paramValue) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putStringSet(paramName, paramValue);
        editor.commit();
    }
}

File: fidouafclient/src/main/java/io/hanko/fidouafclient/util/Preferences.kt
package io.hanko.fidouafclient.util;

import android.content.Context
import android.content.SharedPreferences

object Preferences {

    const val FINGERPRINT_PREFERENCE = "FidoASMPreferences"
    const val LOCKSCREEN_PREFERENCE = "FidoLockscreenPreferences"
    const val PREFERENCE = "FIDO_ASM_PREFERENCE"

    fun create(c: Context, name: String): SharedPreferences {
        return c.getSharedPreferences(name, Context.MODE_PRIVATE)
    }

    fun setParam(preferences: SharedPreferences, paramName: String, paramValue: String) {
        val editor = preferences.edit()
        editor.putString(paramName, paramValue)
        editor.apply()
    }

    fun getParam(preferences: SharedPreferences, paramName: String): String? {
        return preferences.getString(paramName, "")
    }

    fun deleteParam(preferences: SharedPreferences, paramName: String) {
        preferences.edit().remove(paramName).apply()
    }
}
